Ginger/Ginger/UserControlsLib/UCTreeView/UCTreeView.xaml.cs

```diff
@@ -199,6 +211,54 @@ private void SetTreeNodeItemChilds(TreeViewItem TVI)
             }
         }
 
+        private List<ITreeViewItem> GetChildrensByPropertyValueFilter(ITreeViewItem ITVI)
+        {
+            List<ITreeViewItem> childrens = new List<ITreeViewItem>();
+            if (PropertyValueFilter != null)
+            {
+                List<string> propertyStrHierarchy = PropertyValueFilter.Item1.ToString().Split('.').ToList();
+                List<PropertyInfo> propertyInfoHierarchy = new List<PropertyInfo>();
+
+                PropertyInfo pInfo = ITVI.Childrens().Where(z => !(z.NodeObject() is RepositoryFolderBase)).ToList()
+                                                     .First().NodeObject().GetType().GetProperty(propertyStrHierarchy[0]);
+                propertyInfoHierarchy.Add(pInfo);
+
+                if (pInfo != null)
+                {
+                    propertyStrHierarchy.Skip(1).ToList().ForEach(z =>
+                    {
+                        pInfo = pInfo.PropertyType.GetProperty(z.ToString());
+                        propertyInfoHierarchy.Add(pInfo);
+                    });
+
+                    ITVI.Childrens().ForEach(y =>
+                    {
+                        if (y.NodeObject() is RepositoryFolderBase)
+                        {
+                            childrens.Add(y);
+                        }
+                        else
+                        {
+                            List<object> objectsList = new List<object> { propertyInfoHierarchy[0].GetValue(y.NodeObject(), null) };
+                            propertyInfoHierarchy.Skip(1).ToList().ForEach(z =>
+                            {
+                                List<object> objectsListSwap = new List<object>();
+                                objectsList.ForEach(c => objectsListSwap.Add(z.GetValue(c, null)));
+                                objectsList = objectsListSwap;
+                            });
+                            if (objectsList.Contains(PropertyValueFilter.Item2.ToString()))
+                            {
+                                childrens.Add(y);
+                            }
+                        }
+                    });
+                }
+                else
+                    childrens = ITVI.Childrens();
+            }
+            return childrens;
+        }
```

Comment:
looks too complex... is it supporting filtering by fields with more than one child like "aaa.bbb.ccc"?
we need simpler function which acts only on single ITreeViewItem (not on all childreans) and return bool true/false if the specific ITreeViewItem fits to filtering creteria or not.